internal/pkg/v1/action/catalog_update_test.go

```diff
@@ -0,0 +1,84 @@
+package action_test
+
+import (
+	"context"
+	"maps"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	olmv1 "github.com/operator-framework/operator-controller/api/v1"
+
+	internalaction "github.com/operator-framework/kubectl-operator/internal/pkg/v1/action"
+	"github.com/operator-framework/kubectl-operator/pkg/action"
+)
+
+var _ = Describe("CatalogUpdate", func() {
+	setupEnv := func(catalogs ...client.Object) action.Configuration {
+		var cfg action.Configuration
+
+		sch, err := action.NewScheme()
+		Expect(err).To(BeNil())
+
+		cl := fake.NewClientBuilder().
+			WithObjects(catalogs...).
+			WithScheme(sch).
+			Build()
+		cfg.Scheme = sch
+		cfg.Client = cl
+
+		return cfg
+	}
+
+	It("fails finding existing catalog", func() {
+		cfg := setupEnv()
+
+		updater := internalaction.NewCatalogUpdate(&cfg)
+		updater.CatalogName = "does-not-exist"
+		cat, err := updater.Run(context.TODO())
+
+		Expect(err).NotTo(BeNil())
+		Expect(err.Error()).To(ContainSubstring("not found"))
+		Expect(cat).To(BeNil())
+	})
+
+	It("fails to handle catalog with unknown source type", func() {
+		cfg := setupEnv(buildCatalog("test", withCatalogSourceType("invalid-type")))
+
+		updater := internalaction.NewCatalogUpdate(&cfg)
+		updater.CatalogName = "test"
+		_, err := updater.Run(context.TODO())
+
+		Expect(err).NotTo(BeNil())
+		Expect(err.Error()).To(ContainSubstring("unrecognized source type"))
+	})
+
+	It("successfully updates catalog", func() {
+		testCatalog := buildCatalog(
+			"testCatalog",
+			withCatalogSourceType(olmv1.SourceTypeImage),
```

Comment:
Can you add a set of labels to the initial catalog to see that we don't lose them during an update?